boards/arm/mps2_an521/mps2_an521.dts

```diff
@@ -72,13 +77,29 @@
 		};
 	};
 
-	sram0: memory@30000000 {
-		compatible = "mmio-sram";
-		reg = <0x30000000 0x1000000>;
+	ssram1_ns: memory@0 {
+			compatible = "mmio-sram";
+			reg = <0x00000000 0x400000>;
 	};
 
-	flash0: flash@10000000 {
-		reg = <0x10000000 0xE000000>;
+	ssram1_s: memory@10000000 {
```

Comment:
We shouldn't describe both the secure and non-secure memory regions.  They are the same physical memory, and this dts is effectively just the 'secure' side, so we should remove the non-secure regions.